### ISSUE TITLE
SA1119 is reported incorrectly in null-conditional expressions

### ISSUE THREAD

**Issue by geoarion** [DotNetAnalyzers/StyleCopAnalyzers#1348]
Given:
```csharp
int number = (myObject?.intValue).GetValueOrDefault();
```

SA1119 suggests removing the parenthesis, however, that generates invalid code, as *Int* does not have the *GetValueOrDefault()* method from the *Nullable<Int>* type.

Note to self: one (cleaner) alternative to avoid manually suppressing the warning is to write the above expression as:
```csharp
int number = myObject?.intValue ?? 0;
```

**Comment by sharwell** [comment]
Interesting catch! @pdelvo will probably want to take this :bug:

**Comment by pdelvo** [comment]
grabbed
